woocommerce/google-listings-and-ads
# Use AppButton with extracted Button component

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/google-listings-and-ads/issues/1765, and partly addresses https://github.com/woocommerce/google-listings-and-ads/issues/1391.

In this PR, we move from using bundled Button component to using extracted Button component from the `@wordpress/components` package, and remove the usage of `@import "node_modules/@wordpress/components/src/button/style";` in `_gutenberg-components.scss`.

All `<Button>` usages are now changed to `<AppButton>`, including the usages in the `external-components` directory. AppButton is essentially a wrapper around Button. In the future, if there are breaking changes in Button, we just need to apply the fix in AppButton and it should work throughout the whole plugin.

In `_gutenberg-components.scss` file, there were some hacks or fixes under `.components-button` rule. I tested them and they are different from the default `@wordpress/components` button style. The video recording below shows the difference of the button styles in WooCommerce Marketing page and in GLA page:

<details>
	<summary>Sample code for testing button styles used in the video below</summary>

```jsx
<AppButton variant="primary" isDestructive disabled>
	primary destructive disabled
</AppButton>
<AppButton variant="tertiary" isDestructive>
	tertiary destructive
</AppButton>
<AppButton variant="tertiary" isDestructive>
	tertiary destructive hover not disabled
</AppButton>
<AppButton
	variant="link"
	href="https://www.example.com"
	isDestructive
>
	link destructive focus
</AppButton>
<AppButton
	variant="link"
	href="https://www.example.com"
	isDestructive
>
	link destructive focus not disabled
</AppButton>
<AppButton variant="link" href="https://www.example.com">
	link
</AppButton>
```
</details>

https://user-images.githubusercontent.com/417342/210098694-5210c7de-77e4-4c9b-a80e-44984e357b05.mov

To maintain UI appearance in GLA, I have moved the above button styles to AppButton CSS. 

For the `components-panel__body-toggle`'s `padding-right` fix, it seems like it is not needed, so I have removed it. It was introduced in [`2228fa5` (#1708)](https://github.com/woocommerce/google-listings-and-ads/pull/1708/commits/2228fa50828c8ebc0ba235b604385e96e954ddc1) (heads up to the PR author @eason9487). Screenshot below shows the panel without the `padding-right` fix, and it appears to be working as expected without UI issue:

![image](https://user-images.githubusercontent.com/417342/210096240-f855d670-474e-42ec-854f-78e25d647cf5.png)

With the button style no longer bundled and needed, this fixes the CSS issue in non-GLA pages mentioned in https://github.com/woocommerce/google-listings-and-ads/issues/1765.

### Screenshots:

Screenshot of WooCommerce Marketing page, without button style conflict from GLA:

![image](https://user-images.githubusercontent.com/417342/210094900-115a6abd-45cb-4d95-b28c-49c53a179d01.png)

### Detailed test instructions:

1. Buttons in GLA should continue to appear the same as before, and work as expected.
2. Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
3. Go to WooCommerce Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
4. The expand/collapse button in "Discover more marketing tools" card should appear as normal size, not in tiny size.

### Additional details:


### Changelog entry

> Dev - Use extracted Button component from @wordpress/components package.
